### ISSUE TITLE
How to annotate type with a Literal

### ISSUE THREAD

**Issue by janwadolowski** [python/mypy#11748]
```
class Disk(HardwareDefinition):
    DISK_STATUS = Literal["ok", "error"]
    DISK_TYPE = Literal["SSD", "HDD"]

    def __init__(
        self,
        type: DISK_TYPE = "HDD",
    ):
        self.type = type
```

I get an error: error: Variable "DEPLOYMENT_STATUS" is not valid as a type
How should I define that Literal to satisfy mypy?

**Comment by 97littleleaf11** [comment]
Could you provide more information about this? I cannot reproduce the error.

**Comment by janwadolowski** [comment]
Hard to say really, when I rewrote this class in a separarate .py outside of project mypy didn't complain. Perhaps it has something to do with my cofig file?

```
[mypy]
python_version = 3.6
warn_return_any = True
warn_unused_configs = True
warn_unused_ignores = True
warn_redundant_casts = True
no_implicit_optional = True
strict_optional = False
check_untyped_defs = True
follow_imports = silent
follow_imports_for_stubs = True
ignore_missing_imports = True
exclude = .*/utests/test_.*.py$

# Per-module options
[mypy-ff_api.*,ff_backend.*]
disallow_untyped_defs = True
```

Otherwise maybe something to do with python/mypy version (3.8 and mypy 0.910)?

**Comment by sobolevn** [comment]
Sorry, I don't see any `DEPLOYMENT_STATUS` in your example. Maybe it is from somewhere else?

**Comment by janwadolowski** [comment]
sorry, I copied a wrong error message (this issue occured multiple time in my code base with the same error every time). This is the one:
 error: Variable "DISK_TYPE" is not valid as a type

sorry for the confusion

**Comment by sobolevn** [comment]
My suggestion: try to move `DISK_TYPE` outside of `class` definition. Why?

It is not really a part of class as regular python values / methods / etc. It is just a type lias. It should be defined on a module level. The best way to do that is:


```python
from typing import Literal, TypeAlias  # or `typing_extensions`, depends on your python ver

DISK_TYPE: TypeAlias = Literal['a', 'b']
```

**Comment by janwadolowski** [comment]
Okey, thanks a lot. I will try that out :)
